app/javascript/pages/account/my-account.tsx

```diff
@@ -1,15 +1,89 @@
 import React from "react"
 import Layout from "../../layouts/Layout"
 import withAppSetup from "../../layouts/withAppSetup"
+import { Card } from "@bloom-housing/ui-seeds"
+import { Icon, t, type UniversalIconType } from "@bloom-housing/ui-components"
+import UserContext from "../../authentication/context/UserContext"
+import { getSignInPath } from "../../util/routeUtil"
 
 interface MyAccountProps {
   assetPaths: unknown
 }
 
+const AccountDashCard = ({
+  title,
+  description,
+  link,
+  icon,
+  removeBottomBorder = false,
+}: {
+  title: string
+  description: string
+  link: string
+  icon: UniversalIconType
+  removeBottomBorder?: boolean
+}) => {
+  return (
+    <a
+      href={link}
+      className="text-black block outline-none focus:underline flex-1 w-full md:w-1/2  h-auto md:h-60"
+    >
+      <Card
+        spacing="md"
+        className={`cursor-pointer flex justify-center items-center text-center h-full md:h-60 rounded-none md:rounded-lg ${
+          removeBottomBorder ? "border-b-0 md:border-b" : ""
+        }`}
+      >
+        <Card.Header className="flex justify-center w-full flex-col items-center">
+          <div
+            className="pb-4 border-blue-500 w-min px-4 md:px-8 mb-6"
+            style={{ borderBottom: "3px solid" }}
+          >
+            <Icon size="xlarge" className="md:hidden block" symbol={icon} />
+            <Icon size="2xl" className="md:block hidden" symbol={icon} />
+          </div>
+          <h1 className="text-xl md:text-2xl">{title}</h1>
+        </Card.Header>
+
+        <Card.Section>
+          <p className="text-sm">{description}</p>
+        </Card.Section>
+      </Card>
+    </a>
+  )
+}
+
 const MyAccount = (_props: MyAccountProps) => {
+  const { profile, loading, initialStateLoaded } = React.useContext(UserContext)
+
+  if (!profile && !loading && initialStateLoaded) {
+    // TODO: Redirect to React sign in page and show a message that user needs to sign in
+    window.location.href = getSignInPath()
+    return null
```

Comment:
Does setting the `window.location.href` execute the redirect or do we need to do something in addition here?